cms-contentful/src/main/java/io/commercetools/sunrise/cms/contentful/ContentfulCmsService.java

```
@@ -32,14 +32,20 @@
     @Override
     public CompletionStage<Optional<String>> get(final List<Locale> locales, final CmsIdentifier cmsIdentifier) {
         return getEntry(cmsIdentifier)
-                .thenApplyAsync(cdaEntryOptional -> cdaEntryOptional
+                .thenApply(cdaEntryOptional -> cdaEntryOptional
                         .flatMap(cdaEntry -> getLocalizedField(locales, cdaEntry, cmsIdentifier.getFieldName())));
     }
 
     private CompletionStage<Optional<CDAEntry>> getEntry(final CmsIdentifier cmsIdentifier) {
-        return fetchEntry(cmsIdentifier).thenApply(cdaArray -> Optional.ofNullable(cdaArray)
-                .filter(e -> e.items() != null && !e.items().isEmpty())
-                .map(e -> (CDAEntry) e.items().get(0)));
+        return fetchEntry(cmsIdentifier).handle((cdaArray, error) -> {
+            if (error != null) {
+                return Optional.empty();
```

In here you should not swallow the error, but let the completion stage throw it. So like the code before.